rules/Php85/Rector/FuncCall/ArrayKeyExistsNullToEmptyStringRector.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Php85\Rector\FuncCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\BinaryOp\Coalesce;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Scalar\String_;
+use Rector\Rector\AbstractRector;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://wiki.php.net/rfc/deprecations_php_8_5#deprecate_using_values_null_as_an_array_offset_and_when_calling_array_key_exists
+ * @see \Rector\Tests\Php85\Rector\FuncCall\RemoveFinfoBufferContextArgRector\RemoveFinfoBufferContextArgRectorTest
+ */
+final class ArrayKeyExistsNullToEmptyStringRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace null key in array_key_exists with empty string',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+        array_key_exists($key, $array);
+        CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+        array_key_exists($key ?? '', $array);
```

Comment:
I prefer check type, if not string, then use `(string)` cast instead, see other rule

https://github.com/rectorphp/rector-src/blob/9b9271aa37dfdd63d2d455dd408ac4e0d69266a4/rules/Php81/Rector/FuncCall/NullToStrictStringFuncCallArgRector.php#L44

Reply:
Since the logic seems similar, I think you can create re-usable service to be used in both `NullToStrictStringFuncCallArgRector` and this rule so it share similar behaviour, just different php version and function.